src/Manager/FieldSolverBase.h

```diff
@@ -62,10 +62,11 @@ using Solver_t = VariantFromConditionalTypes<CGSolver_t<T, Dim>, FFTSolver_t<T,
 namespace ippl {
   template <typename T, unsigned Dim>
   class FieldSolverBase {
```

Comment:
I do not yet see the purpose of this abstract base class.

Reply:
I wanted to remove FieldSolver from the list of template arguments of PicManager. We need a type here to make shared pointer fsolver_m out of it. However, the FieldSolver class is defined in alpine directory. I find two solutions:

1. We can create a base class for the field solver in src/Manager/. (FieldSolverBase) with very basic member variables. Then, use it to create fsolver_m pointer in pic-manager, and complete/extend it in the FieldSolver class in alpine/ directory by inheritance.

2. In src/Manager/PicManager.h we can include the field solver class from alpine/FieldSolver.hpp and use the class FieldSolver as a type directly to create fsolver_m pointer.

I implemented the 1st solution, which doesn't require including code from alpine/. directory. 
Does it make sense?

Reply:
Yes, It is a good idea to remove the field solver from the list of template arguments. No, the second solution is definitely not a good idea. I agree with your choice.

Reply:
@matt-frey @srikrrish Do you agree on removing FieldSolver and its base class entirely? Should I implement its variables/methods in the AlpineManager?